lib/Document/Adapter/Ghostscript.php

```diff
@@ -210,14 +210,18 @@ public function saveImage(string $imageTargetPath, $page = 1, $resolution = 200)
     /**
      * {@inheritdoc}
      */
-    public function getText(?int $page = null, ?Asset\Document $asset = null)
+    public function getText(?int $page = null, ?Asset\Document $asset = null, ?string $path = null)
     {
         try {
             if (!$asset && $this->asset) {
                 $asset = $this->asset;
             }
 
-            $path = $asset->getLocalFile();
+            if(!$path || !file_exists($path)) {
+                if(self::isFileTypeSupported($asset->getFilename())) {
+                    $path = $asset->getLocalFile();
+                }
+            }
```

Comment:
@Corepex I guess we should throw an exception here in case `$path` is still null.